src/app/workflow/workflow.component.ts

```diff
@@ -398,15 +398,8 @@ export class WorkflowComponent extends Entry<WorkflowVersion> implements AfterVi
           this.updateWorkflowUrl(this.workflow);
         },
         (error) => {
-          const workflowOrgRegex = /\/workflows\/.+/;
-          const workflowRegex = /\/workflows\/(github.com)|(gitlab.com)|(bitbucket.org)\/.+/;
-          const gitHubAppToolRegex = /\/containers\/(github.com)\/.+/;
-          if (
-            workflowOrgRegex.test(this.resourcePath) ||
-            workflowRegex.test(this.resourcePath) ||
-            gitHubAppToolRegex.test(this.resourcePath)
-          ) {
-            this.router.navigate(['page-not-found']);
+          if (!this.workflow) {
```

Comment:
I think it's better to check that the error code is 404 (not found) instead like the container component does

https://github.com/dockstore/dockstore-ui2/blob/fa61caefdc988edf89e0b3170fc252658914d81c/src/app/container/container.component.ts#L263

Reply:
I think checking for a 404 error code causes the [checkerWorkflowFromTools](https://cloud.cypress.io/projects/1ya4vj/runs/33045/overview/d72c1f2b-13e9-45bc-9371-6ed88572d7bd?roarHideRunsWithDiffGroupsAndTags=1) integration test to fail? At 0:40 of the [failed test run](https://cloud.cypress.io/projects/1ya4vj/runs/33045/overview/d72c1f2b-13e9-45bc-9371-6ed88572d7bd/video?roarHideRunsWithDiffGroupsAndTags=1), viewing the parent tool from the checker workflow displays "Page Not Found" instead of the tool entry. It seems like it incorrectly entered the error function's `if (error.status === 404)` block (instead of the error `else` block, which passing tests typically appear to enter -- at 0:43 of this [successful run](https://cloud.cypress.io/projects/1ya4vj/runs/33056/test-results/580ebfdf-da75-41b2-8710-ef6df9913223/video?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D), the page shows "The workflow linked to could not be found!")